JavaScript/sign-script.js

```diff
@@ -0,0 +1,77 @@
+// const emailInput = document.getElementsByClassName('email')[0];
+// const errorMessage = document.getElementsByClassName('error-message')[0];
+const emailInput = document.querySelectorAll('.email')[0];
+const passInput = document.querySelectorAll('.password')[0];
+const passConfirmInput = document.querySelectorAll('.password-confirm')[0];
+const errorMail = document.querySelectorAll('.error-mail')[0];
+const errorPass = document.querySelectorAll('.error-pass')[0];
+const errorPassConfirm = document.querySelectorAll('.error-passconfirm')[0];
+const eyeImagePass = document.querySelector('#eyeImagePass'); 
+const eyeImageConfirm = document.querySelector('#eyeImageConfirm'); 
+const pattern = /^[A-Za-z0-9_]+[A-Za-z0-9]*[@]{1}[A-Za-z0-9]+[A-Za-z0-9]*[.]{1}[A-Za-z]{1,3}$/;
+let link = '/folder';
+
+function checkEmail(emailType) {
+  if (emailInput.value === '') {
+    errorMail.textContent = '이메일을 입력해주세요.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (!pattern.test(emailType)) {
+    errorMail.textContent = '올바른 이메일 주소가 아닙니다.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (emailInput.value !== '') {
+    errorMail.textContent = '이메일을 확인해주세요.';
+    emailInput.classList.add('error'); // 에러 클래스 추가
+  } else if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
```

Comment:
```suggestion
  } else if (emailInput.value !== '') {
    errorMail.textContent = '이메일을 확인해주세요.';
    emailInput.classList.add('error'); // 에러 클래스 추가
  } else if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
```

여기 로직이 정상적으로 돌아가나요?
지금 보기에는 emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101' 조건에 들어오기 전에 emailInput.value !== ''여기서 걸릴 것 같아 보이네요...? 순서 변경이 필요해보입니다.

Reply:
여기가 좀 이상하다고 생각했는데 계속 봐도 왜 걸리는지 이해가 되지 않습니다.. 두 코드 모두 따로 동작하는 것 아닌가요? 각각 if문으로 따로 떼서 얼리 리턴을 적용하면 상관없어지나요?